templates/quickstart-hashicorp-vault.template

```diff
@@ -312,364 +578,397 @@ Resources:
+                  - S3Region: !If [UsingDefaultBucket, !Ref 'AWS::Region', !Ref QSS3BucketRegion]
+                    S3Bucket: !If [UsingDefaultBucket, !Sub '${QSS3BucketName}-${AWS::Region}', !Ref QSS3BucketName]
               mode: '000700'
               owner: root
               group: root
             "/etc/systemd/system/vault.service":
               source:
-                Fn::Sub:
-                - https://s3.amazonaws.com/${QSS3BucketName}/${QSS3KeyPrefix}scripts/vault.service
-                - QSS3BucketName:
-                    Ref: QSS3BucketName
-                  QSS3KeyPrefix:
-                    Ref: QSS3KeyPrefix
+                !Sub
+                  - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}scripts/vault.service'
+                  - S3Region: !If [UsingDefaultBucket, !Ref 'AWS::Region', !Ref QSS3BucketRegion]
+                    S3Bucket: !If [UsingDefaultBucket, !Sub '${QSS3BucketName}-${AWS::Region}', !Ref QSS3BucketName]
               mode: '000700'
               owner: root
               group: root
           commands:
             01_chmod_and_chown_vault:
-              command: chmod 0755 /usr/bin/vault && chown root:root /usr/bin/vault
+              command: chmod 0755 /usr/bin/vault && chown vault:vault /usr/bin/vault
         run_vault:
           commands:
             01_register_vault_binary:
-              command: service vault start
+              command: systemctl daemon-reload && systemctl enable vault && systemctl start vault
+    Type: AWS::AutoScaling::LaunchConfiguration
     Properties:
-      InstanceType:
-        Ref: VaultInstanceType
-      KeyName:
-        Ref: KeyPair
-      NetworkInterfaces:
-      - DeleteOnTermination: true
-        DeviceIndex: '0'
-        SubnetId:
-          Ref: PrivateSubnet1ID
-        GroupSet:
-        - Ref: VaultSecGroup
+      LaunchConfigurationName: !Sub "VaultLC-${AWS::StackName}"
+      InstanceType: !Ref VaultInstanceType
+      KeyName: !Ref KeyPairName
+      SecurityGroups: 
+      - !Ref VaultSecGroup
       ImageId:
         Fn::FindInMap:
         - AWSAMIRegionMap
-        - Ref: AWS::Region
-        - US1604HVM
+        - !Ref 'AWS::Region'
+        - !FindInMap
+          - LinuxAMINameMap
+          - !Ref VaultAMIOS
+          - Code
       BlockDeviceMappings:
       - DeviceName: "/dev/sda1"
         Ebs:
           VolumeSize: 20
           VolumeType: gp2
+      IamInstanceProfile: !Ref "VaultInstanceProfile" 
       UserData:
-        Fn::Base64:
-          Fn::Join:
-          - ''
-          - - "#!/bin/bash -x\n"
-            - 'apt-get -y update
-
-'
-            - 'apt-get -y install  python-setuptools
-
-'
-            - 'easy_install https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz
+        Fn::Base64: !Sub
+          - |
+            #!/bin/bash -x
+            
+            # Install git
+            apt-get update
+            apt-get install -y git
```

Comment:
Can you also install `jq` package?

Reply:
I have installed JQ at an earlier point. I will try and consolidate package installations to a single point to make this less confusing.